xapian-core/tests/termgentest.cc

```diff
@@ -735,7 +755,22 @@ static bool test_termgen1()
 	    } else if (strncmp(o, "some", 4) == 0) {
 		o += 4;
 		termgen.set_stemming_strategy(termgen.STEM_SOME);
-	    } else if (strncmp(o, "prefix=", 7) == 0) {
+	    } else if(strncmp(o, "stop=", 5) == 0) {
+               o += 7;
+               vector<string> x = {"the", "a", "an"};
+               Xapian::SimpleStopper *stopper = new Xapian::SimpleStopper(x.begin(), x.end());
+               Xapian::Stopper *stop = static_cast<Xapian::Stopper *>(stopper);
+               termgen.set_stopper(stop);
```

Comment:
The cast here should be unnecessary - the compiler can implicitly convert a pointer to a derived class into a pointer to the base class.

However, you do have a memory leak here - you need to release ownership of the `Stopper` object so that the `TermGenerator` object will delete it later:

```
termgen.set_stopper(stopper->release());
```